staging/src/k8s.io/apimachinery/pkg/runtime/serializer/json/json.go

```diff
@@ -32,6 +33,58 @@ import (
 	utilyaml "k8s.io/apimachinery/pkg/util/yaml"
 )
 
+var (
+	// Private copy of jsoniter.ConfigCompatibleWithStandardLibrary to try to shield against possible mutations
+	// from outside. Still does not protect from package level jsoniter.Register*() functions - someone calling them
+	// in some other library will mess with every usage of the jsoniter library in the whole program.
+	// See https://github.com/json-iterator/go/issues/265
+	configCompatibleWithStandardLibrary = jsoniter.Config{
+		EscapeHTML:             true,
```

Comment:
this needs a test to detect drift.

Reply:
@thockin 
> Can we start with their struct and make a deepcopy or something?

We cannot make a copy because `jsoniter.ConfigCompatibleWithStandardLibrary` is an interface.

> Can we get them (upstream) to have an immutable baseline object and then a mutable one that derives from that. Then we can skip th emutable one, but stay in sync with the base?

I have tried to communicate the issue but failed. See https://github.com/json-iterator/go/issues/265.

I don't want to boil the ocean here, just want to fix the bug. I can reduce the scope of this PR to just add the missing error handling line. WDYT?

Reply:
I mean, it is sad, but... what else can I do?

Reply:
Can we get them to expose the pre-Froze() struct so we can copy that and Froze() it ourself?

Reply:
Rather than race to get this in 1.11, is it a big deal to try to get good resolution to this issue and then aim for 1.12 ?  Is there any real issue being triggered?

Reply:
I think this can be delayed until later.